src/common/com/intellij/plugins/haxe/ide/module/HaxeModuleBuilder.java

```diff
@@ -51,6 +67,76 @@ public void moduleCreated(@NotNull Module module) {
     final CompilerModuleExtension model = (CompilerModuleExtension)CompilerModuleExtension.getInstance(module).getModifiableModel(true);
     model.setCompilerOutputPath(model.getCompilerOutputUrl());
     model.inheritCompilerOutputPath(false);
+    createHelloWorldIfEligible(module);
+
     model.commit();
   }
+
+  private void createHelloWorldIfEligible(Module module) {
+    ModuleRootManager rootManager = ModuleRootManager.getInstance(module);
+    VirtualFile[] srcDirs = rootManager.getSourceRoots();
+    VirtualFile[] rootDirs = rootManager.getContentRoots();
+    if(rootDirs.length != 1 || srcDirs.length != 1) {
+      return;
+    }
+
+    VirtualFile root = rootDirs[0];
+    VirtualFile src = srcDirs[0];
+
+    if(src.getChildren().length != 0) {
+      return;
+    }
+    for(VirtualFile item:root.getChildren()) {
+      if(item.getExtension() == "hxml") {
+        return;
+      }
+    }
+
+
+    ApplicationManager.getApplication().runWriteAction(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          VirtualFile mainHx = src.createChildData(this, "Main.hx");
+          String mainHxSource = "class Main {\n"
+            + "    static public function main() {\n"
+            + "        trace(\"Hello, world!\");\n"
+            + "    }\n"
+            + "}\n";
+          mainHx.setBinaryContent(mainHxSource.getBytes(StandardCharsets.UTF_8));
+
+          VirtualFile buildHxml = root.createChildData(this, "build.hxml");
+          String buildHxmlSource = "-cp src\n"
+            + "-D analyzer-optimize\n"
+            + "-main Main\n"
+            + "--interp";
+          buildHxml.setBinaryContent(buildHxmlSource.getBytes(StandardCharsets.UTF_8));
+
+          createDefaultRunConfiguration(module, buildHxml.getPath());
+
+          FileEditorManager editorManager = FileEditorManager.getInstance(module.getProject());
+          editorManager.openFile(mainHx, true);
+
+        } catch (IOException e) {
+        }
+      }
+    });
+    LOG.debug("Hey");
```

Comment:
Probably don't want to leave this log in.  And since it's the only log line in the file, we don't need the LOG variable created at the top of this class.